microcosm/loaders/secrets_file.py

```diff
@@ -0,0 +1,58 @@
+"""
+Load configuration from a settings file.
+
+The settings file is identified using a `FOO_SETTINGS` environment variable
+and consumed using a customizable load function (default: json).
+
+"""
+from json import loads
+from os import environ
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Mapping,
+    Optional,
+)
+
+from inflection import underscore
+
+from microcosm.metadata import Metadata
+
+
+def get_config_filename(metadata: Metadata) -> Optional[str]:
+    """
+    Derive a configuration file name from the FOO_SECRETS
+    environment variable.
+
+    """
+    envvar = "{}__SECRETS".format(underscore(metadata.name).upper())
+    try:
+        return environ[envvar]
+    except KeyError:
+        return None
+
+
+def _load_from_file(metadata: Metadata, load_func: Callable[[str], Mapping[str, str]]) -> Dict[Any, Any]:
+    """
+    Load configuration from a file.
+
+    The file path is derived from an environment variable
+    named after the service of the form FOO_SECRETS.
+
+    """
+    config_filename = get_config_filename(metadata)
+    if config_filename is None:
+        return dict()
+
+    with open(config_filename, "r") as file_:
+        data = load_func(file_.read())
+        return dict(data)
```

Comment:
What's the expected format of the data in the file?

Reply:
this matches the behavior of the normal config file loader, so the deserialization is performed by the passed in callable `load_func`, this is defined in `load_from_json_secrets_file` as `loads` from the json package